File: truyenyy/src/page.lua
local url = ...
local list = {}
local new_url = text:replace(url, "truyenyy.com", "truyenyy.vip")
new_url = text:replace(new_url, "truyenyy.vn", "truyenyy.vip")
local doc = http:get(new_url .. "/danh-sach-chuong"):timeout(0):html()

local page = num:to_int(regexp:find_last(doc:html(), "\\?p=\\d+\\s*\">\\s*(\\d+)\\s*<"), 1)

for i = 1, page do
    table.insert(list, text:replace(url, "truyenyy.vn", "truyenyy.vip") .. "/danh-sach-chuong/?p=" .. i)
end
return response:success(list)

File: truyenyy/src/page.js
function execute(url) {
    url = url.replace("truyenyy.com", "truyenyy.vip")
        .replace("truyenyy.vn", "truyenyy.vip");
    let response = fetch(url + "/danh-sach-chuong");
    if (response.ok) {
        let doc = response.html();

        let pageList = [];
        let isMobile = doc.select("meta[name=mobile-web-app-capable]").attr("content") === "yes";
        if (isMobile) {
            var page = doc.select(".novel-detail > .cell-box").last().select("a");
            if (page && page.size() > 0)
                page = /p=(\d+)/.exec(page.last().attr("href"))[1];
            else page = 1;

            for (let i = 1; i <= page; i++)
                pageList.push(url + "/danh-sach-chuong/?p=" + i);
        } else {
            var page = doc.select(".pagination").last().select("a");
            if (page && page.size() > 0)
                page = parseInt(page.get(page.size() - 2).text());
            else page = 1;

            for (let i = 1; i <= page; i++)
                pageList.push(url + "/danh-sach-chuong/?p=" + i);
        }
        return Response.success(pageList);
    }

    return null;
}
